trl/models/modeling_base.py

```diff
@@ -386,18 +386,18 @@ def _get_checkpoint_from_hub(
     @classmethod
     def _get_current_device(cls):
         r"""
-        Get the current device. For GPU, we return the local process index using LOCAL_RANK
+        Get the current device. For GPU, we return the local process index using PartialState
         object to handle corner cases when running scripts in distributed environments.
 
         Returns:
             current_device (`Union[int, str]`):
                 The current device.
         """
-        local_rank = int(os.environ.get('LOCAL_RANK', '0'))
+        state = PartialState()
         if is_xpu_available():
-            return f"xpu:{local_rank}"
+            return f"xpu:{state.local_process_index}"
         else:
-            return local_rank if torch.cuda.is_available() else "cpu"
+            return state.local_process_index if torch.cuda.is_available() else "cpu"
```

Comment:
This whole thing can be replaced with `state.local_process_index` unless `xpu` is needed.

Reply:
I'd say that can be in another PR then just to keep things isolated a bit :) LG2M then!